src/components/editor/YextEntityFieldSelector.tsx

```diff
@@ -21,16 +28,100 @@ export type RenderYextEntityFieldSelectorProps<T extends Record<string, any>> =
     filter: RenderEntityFieldFilter<T>;
   };
 
-const SUPPORTED_CONSTANT_VALUE_TYPES: EntityFieldTypes[] = [
-  "type.string",
-  "type.phone",
-];
-const shouldDisplayConstantValueField = (typeFilter: EntityFieldTypes[]) => {
-  for (const constantValueType of SUPPORTED_CONSTANT_VALUE_TYPES) {
-    if (typeFilter.includes(constantValueType)) {
-      return true;
+const TEXT_CONSTANT_CONFIG: TextField = {
+  type: "text",
+};
+
+const IMAGE_CONSTANT_CONFIG: CustomField<ImageType> = {
+  type: "custom",
+  render: ({ onChange, value }) => {
+    return (
+      <>
+        <FieldLabel
+          label={"Alternate Text"}
+          className="ve-inline-block ve-pt-4"
+        >
+          <AutoField
+            field={{ type: "text" }}
+            value={value.alternateText}
+            onChange={(fieldValue) => {
+              onChange({
+                ...value,
+                alternateText: fieldValue,
+              });
+            }}
+          />
+        </FieldLabel>
+        <FieldLabel label={"Height"} className="ve-inline-block ve-pt-4">
+          <AutoField
+            field={{ type: "number" }}
+            value={value.height}
+            onChange={(fieldValue) => {
+              onChange({
+                ...value,
+                height: fieldValue,
+              });
+            }}
+          />
+        </FieldLabel>
+        <FieldLabel label={"Width"} className="ve-inline-block ve-pt-4">
+          <AutoField
+            field={{ type: "number" }}
+            value={value.width}
+            onChange={(fieldValue) => {
+              onChange({
+                ...value,
+                width: fieldValue,
+              });
+            }}
+          />
+        </FieldLabel>
+        <FieldLabel label={"URL"} className="ve-inline-block ve-pt-4">
+          <AutoField
+            field={{ type: "text" }}
+            value={value.url}
+            onChange={(fieldValue) => {
+              onChange({
+                ...value,
+                url: fieldValue,
+              });
+            }}
+          />
+        </FieldLabel>
+      </>
+    );
+  },
+};
+
+const TYPE_TO_CONSTANT_CONFIG: Record<string, Field<any>> = {
+  "type.string": TEXT_CONSTANT_CONFIG,
+  "type.phone": TEXT_CONSTANT_CONFIG,
+  "type.image": IMAGE_CONSTANT_CONFIG,
+};
+
+/**
+ * Returns the constant type configuration if all types match
+ * @param typeFilter
+ */
+const returnConstantFieldConfig = (
+  typeFilter: EntityFieldTypes[]
+): Field | undefined => {
+  let fieldConfiguration: Field | undefined;
+  for (const entityFieldType of typeFilter) {
+    const mappedConfiguration = TYPE_TO_CONSTANT_CONFIG[entityFieldType];
+    if (!mappedConfiguration) {
+      console.log(`No mapped configuration for ${entityFieldType}`);
```

Comment:
I think we might want these to be through dev logger. They display on the published and will be printed for every custom field type used